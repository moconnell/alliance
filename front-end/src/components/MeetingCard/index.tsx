import Meeting from "../../types/meeting";
import {
  DateText,
  CardContainer,
  MeetingDescription,
  Attendee,
} from "./styles";

interface MeetingCardProps {
  meeting: Meeting;
}

const MeetingCard = ({
  meeting: { date, endDate, attendee, description },
}: MeetingCardProps) => (
  <CardContainer>
    <DateText>
      {`${date.toLocaleString({timeStyle: "short"})}-${endDate.toLocaleString({timeStyle: "short"})}`}
    </DateText>
    <Attendee>{attendee}</Attendee>
    <MeetingDescription>{description}</MeetingDescription>
  </CardContainer>
);

export default MeetingCard;
