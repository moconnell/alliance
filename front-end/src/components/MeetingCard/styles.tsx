import styled from "styled-components";

export const CardContainer = styled.div`
  max-width: 200px;
  width: 196px;
  height: 170px;
  max-height: 200px;
  display: flex;
  justify-content: left;
  flex-wrap: wrap;
  background: #ff3864;
  border: 1px solid #ff3864;
  box-sizing: border-box;
  border-radius: 6px;
  padding: 6px 8px;
  margin-bottom: 6px;
`;
export const Attendee = styled.span`
  font-family: Inter;
  font-style: normal;
  font-weight: 600;
  font-size: 1rem;
  color: #fff;
  overflow: hidden;
  max-width: 230px;
  white-space: nowrap;
  text-overflow: ellipsis;
`;

export const MeetingDescription = styled.span`
  font-family: Inter;
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 15px;
  color: #fff;
`;

export const DateText = styled.span`
  font-family: Inter;
  font-style: normal;
  font-size: 0.9rem;
  line-height: 28px;
  color: #fff;
`;
