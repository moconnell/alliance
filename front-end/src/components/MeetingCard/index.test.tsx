import { render } from "@testing-library/react";
import { DateTime } from "luxon";
import MeetingCard from "./index";

describe("<MeetingCard />", () => {
  it("should render correctly", async () => {
    const meeting = {
      date: DateTime.fromISO("2022-02-02T20:00:00").setLocale("en-gb"),
      endDate: DateTime.fromISO("2022-02-02T20:30:00").setLocale("en-gb"),
      attendee: "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266",
      description: "30 minute meeting",
    };
    const { asFragment } = render(<MeetingCard meeting={meeting} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
