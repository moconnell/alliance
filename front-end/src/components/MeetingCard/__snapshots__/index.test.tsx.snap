// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MeetingCard /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ lhfvPN"
  >
    <span
      class="sc-hKMtZM cKEwGb"
    >
      20:00-20:30
    </span>
    <span
      class="sc-gsnTZi cMktXF"
    >
      0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266
    </span>
    <span
      class="sc-dkzDqf ezVsXa"
    >
      30 minute meeting
    </span>
  </div>
</DocumentFragment>
`;
